### ISSUE TITLE
projection.copy?

### ISSUE THREAD

**Issue by mbostock** [d3/d3-geo#123]
It’d be nice to have a method to create a copy of a projection with identical settings (that, or change projections to be immutable, and every method that modifies the projection instead returns a new projection).

**Comment by tinius** [comment]
+1 for immutable projections. I might even be able to help with this

**Comment by Fil** [comment]
Should we try and do this for d3-geo@2 d3-geo-projection@3 d3-geo-polygon@2 ?

I'm really not sure about immutable projections — they would mean rewriting almost all map interactions everywhere; but projection.copy() would be super useful in many instances.

Related (maybe), I very often find that I would need the projections to carry some metadata (for example & at the very least, a name; but also a flag indicating if it needs clipping to the sphere, etc).

**Comment by nirazul** [comment]
This would come in very handy when implementing rotation calculations as I currently need to set the rotation to `[0, 0]` and reset it to the original state after having calculated a point with [`attitude`](https://www.npmjs.com/package/attitude).

For example:
```
function getProjectedPoint(projection, event) {
  const originalRotation = projection.rotate()
  const pt = pointers(event, event.sourceEvent.target)[0]
  const result = projection.rotate([0, 0]).invert(pt)

  projection.rotate(originalRotation)

  return result
}
```

**Comment by Fil** [comment]
Agree, but I don't have the beginning of an idea how to implement this—projections are a very complex object and there's no easy "deep clone" operation. Maybe I'm missing something? Contributions are welcome.
